BranchMetrics/android-branch-deep-linking-attribution
# Sdk 855/oaid

## Reference
SDK-SDK-855 -- OAID support.

## Description
Huawei cannot use GMS (Google Mobile Services), similarly to Fire OS, so they have an equivalent to Google's advertising ID called OAID. When we detect that the phone is using HMS instead of GMS, we need to use that instead.

## Testing Instructions
Need a Huawei Mate 30 (on my desk), run run this app https://github.com/bklastaitis-branch/HuaweiOAIDTest on it, see the payload that contains OAID.

## Risk Assessment [`HIGH` || `MEDIUM` || `LOW`]
LOW (Normal Android OS is unaffected)


## Reviewer Checklist (To be checked off by the reviewer only)

- Correctness & Style
